src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/ObjectModelConverters.cs

```diff
@@ -124,14 +132,23 @@ private static void CopyVSTestProviderProperties(TestNode testNode, TestCase tes
         {
             testNode.Properties.Add(new SerializableKeyValuePairStringProperty("vstest.original-executor-uri", originalExecutorUri.AbsoluteUri));
         }
+
+        if (!RoslynString.IsNullOrEmpty(testCase.CodeFilePath) &&
+            clientInfo.Id == WellKnownClients.VisualStudio &&
+            Version.TryParse(clientInfo.Version, out Version? clientVersion) &&
```

Comment:
maybe the client info could be wrapped in some compatibility service that implements the comparison, so it can use more concrete type when available and cache, so we don't have to re-parse it every time and have 10 different ways of handling the version comparison?